src/Language/Haskell/Liquid/Bare/Lookup.hs

```diff
@@ -162,35 +194,39 @@ lookupGhcVar :: GhcLookup a => a -> BareM Var
 lookupGhcVar x
   = do env <- gets varEnv
        case L.lookup (symbol x) env of
-         Nothing -> lookupGhcThing "variable" fv x
+         Nothing -> lookupGhcThing "variable" fv (Just varName) x `catchError` \_ ->
+                    lookupGhcThing "variable or data constructor" fv (Just dataName) x
          Just v  -> return v
   where
     fv (AnId x)                   = Just x
     fv (AConLike (RealDataCon x)) = Just $ dataConWorkId x
     fv _                          = Nothing
 
 
-lookupGhcTyCon   ::  GhcLookup a => a -> BareM TyCon
-lookupGhcTyCon s = lookupGhcThing err ftc s `catchError` \_ ->
-                         lookupGhcThing err fdc s
+lookupGhcTyCon   ::  GhcLookup a => String -> a -> BareM TyCon
+lookupGhcTyCon src s = do 
+  lookupGhcThing err ftc (Just tcName) s  `catchError` \_ ->
+   lookupGhcThing err fdc (Just tcName) s 
   where
     -- s = trace ("lookupGhcTyCon: " ++ symbolicString _s) _s
     ftc (ATyCon x)
       = Just x
     ftc _
       = Nothing
 
-    fdc (AConLike (RealDataCon x)) | isJust $ promoteDataCon_maybe x
+    fdc (AConLike (RealDataCon x)) | showPpr x == "GHC.Types.IO"  -- getUnique x `hasKey` ioTyConKey 
```

Comment:
The uniques do not match, because the data constructor is returned, instead of the type constructor. I suspect this is a ghc bug...